### PR TITLE
Separate remaining moves estimator from time allocation curve

### PR DESCRIPTION
**Pull request by borg323**
This is my effort to separate the time allocation curve from the remaining moves estimator. The existing estimator parameters are renamed to `GameMidpointMove` and `GameMovesSteepness` (still hidden) and the defaults are changed as in <https://github.com/LeelaChessZero/lc0/issues/869#issuecomment-521695618> to serve as an estimator of just the remaining moves, one that can be straightforwardly replaced by the moves left head (https://github.com/LeelaChessZero/lc0/pull/961) if available.
This change results in a flat time allocation that seems to lose strength in most cases, so the resulting time allocation is multiplied by a new time curve to use more time early in the game. This new time curve reuses the old parameter names together with two new ones: `TimeMidpointMove`, `TimeSteepness`, `TimeFactor` and `TimeFactorSlope`. Those define the following shape:
time_curve

@kiudee was kind enough to do a first optimization pass over those parameters to get strong defaults:

I consider the new time curve shape (and parameter names) provisional, so alternatives and any other suggestions are welcome.

### CODE REVIEW

**Comment by zz4032** [comment]
Gauntlet vs. SF:
```
   # ENGINE               :  RATING  ERROR  CFS(%)   GAMES  DRAWS(%)
   1 stockfish            :     0.0   ----    63.1    3000      53.6
   2 lc0-master           :    -2.0   11.8    72.4    1500      53.2
   3 lc0-master_PR1065    :    -6.8   11.3     ---    1500      53.9

White advantage = 97.7 +/- 3.9
Draw rate (equal opponents) = 58.7 % +/- 0.9
```
- default time control settings
- 3-move book
- 6-man TBs + adjudication
- network: Id590976
- tc: 30s/game+0.3s/move (Leela), 120s/game+1.2s/move (SF)
- average time/move: 0.8s/move (Leela)
- `concurrency 8` in Cutechess

**Comment by mooskagh** [comment]
I remember that the current time constants were tuned on short TC games too, and then they worked very poor at longer games (basically all time was forced to drain on obvious moves between moves 50 and 60).
We need to make sure somehow that there won't be similar problems now.

File: src/mcts/stoppers/factory.cc [hunk 1]
```diff
@@ -55,17 +55,30 @@ const OptionId kSlowMoverId{
...
-    "The move where the time budgeting algorithm guesses half of all "
-    "games to be completed by. Half of the time allocated for the first move "
-    "is allocated at approximately this move."};
-const OptionId kTimeSteepnessId{
-    "time-steepness", "TimeSteepness",
-    "\"Steepness\" of the function the time budgeting algorithm uses to "
-    "consider when games are completed. Lower values leave more time for "
-    "the endgame, higher values use more time for each move before the "
-    "midpoint."};
+const OptionId kTimeFactorId{"time-factor", "TimeFactor",
```
Comment (mooskagh) [inline src/mcts/stoppers/factory.cc]: `initial-slowmover`?
At least the help needs to be changed. s/start allocations/first moves/?

File: src/mcts/stoppers/factory.cc [hunk 2]
```diff
@@ -55,17 +55,30 @@ const OptionId kSlowMoverId{
...
+                             "Like slowmover, but for the start allocations."};
+const OptionId kTimeFactorSlopeId{"time-factor-slope", "TimeFactorSlope",
```
Comment (mooskagh) [inline src/mcts/stoppers/factory.cc]: Hm...
Maybe
* initial-timebudget
* time-growth-slope
* time-growth-to-decay-move
* time-decay-slope

Meh.. :\

Comment (mooskagh) [reply src/mcts/stoppers/factory.cc]: (feel free to push back on suggestions!)

**Comment by borg323** [comment]
Superseded by #1195

**Comment by mooskagh** [comment]
Actually, the idea is still relevant, it's just that time management code is moved around.

**Comment by borg323** [comment]
#1195 separated the remaining moves estimator from the time management, so it achieves the objective of this. The curve I used here was just an attempt to approximate the legacy time management. The only thing still pending is the default values for the remaining moves estimator, but this is covered in <https://github.com/LeelaChessZero/lc0/issues/869#issuecomment-521695618>.

**Pull request closed without merging**